connectors/sources/sharepoint.py

```diff
@@ -284,7 +284,25 @@ async def _api_call(self, url_name, url="", **url_kwargs):
                         yield await result.json()
                     break
             except Exception as exception:
-                if isinstance(
+                if retry > self.retry_count:
+                    raise exception
+                logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+                )
+                retry += 1
+
+                retry_after = RETRY_INTERVAL**retry
+
+                if getattr(exception, "status", None) == 429:
+                    retry_after = int(
+                        exception.headers.get(  # pyright: ignore
+                            "Retry-After", RETRY_INTERVAL
+                        )
+                    )
+                    logger.warning(
+                        f"Rate limit exceeds. Retrying after {retry_after} seconds."
+                    )
```

Comment:
Why would we retry if it's not 429?

Reply:
There are some errors such as ServerDisconnectedError which can be fixed in after few retries

Reply:
It should be explicit then - catch `ServerDisconnectedError` and handle it instead of handling `Exception`.